### PR TITLE
boards: hifive1: Added flash runner

### PR DESCRIPTION
**Pull request by timbz**
Signed-off-by: Timon Baetz <timon.baetz@gmail.com>

### CODE REVIEW

**Comment by timbz** [comment]
I am able to flash my hifive1 on my mac using a custom zephyr toolchain configured to use the prebuilt GCC form the sifive homepage. I also have the riscv openocd in my PATH. 
On Linux with the Zephyr SDK you probably have to add a -DOPENOCD=<path_to_riscv_openocd> to the cmake command. Something like 
`cmake -GNinja -DBOARD=hifive1 -DOPENOCD=<openocd_path> ..`

**Review by nategraff-sifive — CHANGES_REQUESTED** [review]
This worked great!

Could update the board documentation for the new process in `boards/riscv32/hifive1/doc/hifive1.rst`?

Thanks again for your contribution!

File: boards/riscv32/hifive1/doc/hifive1.rst
Comment (dbkinder) [inline boards/riscv32/hifive1/doc/hifive1.rst]: Can you write how to verify you have the "correct" version of OpenOCD?  Is there a version number you can check?

**Review by dbkinder — APPROVED** [review]
Looks good, thanks.  One question though...

**Review by dbkinder — DISMISSED** [review]
whoops.  Wanted to wait to approve after the question got answered :)

File: boards/riscv32/hifive1/doc/hifive1.rst
Comment (nategraff-sifive) [reply boards/riscv32/hifive1/doc/hifive1.rst]: My own recommendation would be to use the latest version of RISC-V OpenOCD distributed [on the SiFive website](https://www.sifive.com/boards).

The steps would be
1. Download the OpenOCD tarball for your OS
2. Extract the tarball
3. Pass `-DOPENOCD=path/to/riscv-openocd/bin/openocd` to cmake

We can also probably get rid of the reference to Freedom E SDK, since the only reason to pull that down is for the OpenOCD configuration, which would now be in Zephyr.

Comment (timbz) [reply boards/riscv32/hifive1/doc/hifive1.rst]: I updated the pull request. Please let me know what you think.

**Review by nategraff-sifive — APPROVED** [review]

File: boards/riscv32/hifive1/doc/hifive1.rst
Comment (timbz) [reply boards/riscv32/hifive1/doc/hifive1.rst]: > Can you write how to verify you have the "correct" version of OpenOCD? Is there a version number you can check?

This is the version output from OpenOCD:
```
# zephyr sdk
$ openocd -v
Open On-Chip Debugger 0.10.0+dev-gea2753a7-dirty (2018-11-02-23:37)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
```
```
# risc-v
$ openocd -v
Open On-Chip Debugger 0.10.0+dev-gcd32bd4 (2018-12-19-01:15)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
```

Unfortunately that doesn't help. I guess you just have to know what you downloaded

Comment (timbz) [reply boards/riscv32/hifive1/doc/hifive1.rst]: @dbkinder is there anything else you want me to change?

**Review by kgugala — APPROVED** [review]

File: boards/riscv32/hifive1/doc/hifive1.rst
Comment (mbolivar) [inline boards/riscv32/hifive1/doc/hifive1.rst]: s/bundeld/bundled/

File: scripts/west_commands/runners/__init__.py
Comment (mbolivar) [inline scripts/west_commands/runners/__init__.py]: Thanks for keeping this in alphabetical order :+1:

**Review by mbolivar — CHANGES_REQUESTED** [review]
Looks good! There is a minor typo in the documentation patch. Sorry I didn't see this before; I need to get myself on the CODEOWNERS file for the new location of these files.

Happy to +1 with that fixed.

File: boards/riscv32/hifive1/doc/hifive1.rst
Comment (timbz) [reply boards/riscv32/hifive1/doc/hifive1.rst]: Fixed and rebased

**Review by mbolivar — APPROVED** [review]

**Review by galak — CHANGES_REQUESTED** [review]
Do we need boards/riscv32/hifive1/support/openocd.cfg?  Shouldn't that be coming from openocd?  The Zephyr sdk 0.10-beta5 includes an updated openocd from upstream that seems to have this file in it.

**Comment by mbolivar** [comment]
> Do we need boards/riscv32/hifive1/support/openocd.cfg? Shouldn't that be coming from openocd?

```
$ find boards -name openocd.cfg  | wc -l
64
```

The usual convention and pattern is for Zephyr boards to provide this. Changing that assumption would be a bigger project. Let's not forget that SDK only covers Linux, for example.

**Comment by galak** [comment]
> The usual convention and pattern is for Zephyr boards to provide this. Changing that assumption would be a bigger project. Let's not forget that SDK only covers Linux, for example.

Sure, but this is duplicating a file identically that is part of openocd.  So not sure what the value of doing that is.

**Comment by mbolivar** [comment]
> Sure, but this is duplicating a file identically that is part of openocd. So not sure what the value of doing that is.

There are a ton of places to get openocd out there. Do they all have this file?

**Comment by mbolivar** [comment]
Further, why are we making an example of this board? If we really want to avoid having these files in Zephyr we should just delete all of them, no?

**Comment by nashif** [comment]
> Sure, but this is duplicating a file identically that is part of openocd. So not sure what the value of doing that is.

which file is this duplicating? This file additional has the gdb stubs, those are not in openocd. This file can also be used to add commands that are not in the default provided by openocd.

**Comment by timbz** [comment]
I updated the pull request yesterday evening and replaced the content of the cgf file with ```source [find board/sifive-hifive1.cfg]```

**Comment by timbz** [comment]
I did some more testing on different platforms. It seems to work fine with both the Zephyr SDK and custom toolchains (with the 9.5 version of the SDK you still need the `-DOPENOCD`parameter)

**Review by galak — APPROVED** [review]

**Pull request merged**
